freqtrade/optimize/hyperopt.py

```diff
@@ -546,10 +546,11 @@ def generate_optimizer(self, raw_params: List[Any], iteration=None) -> Dict:
 
         )
         return self._get_results_dict(backtesting_results, min_date, max_date,
-                                      params_dict, params_details)
+                                      params_dict, params_details,
+                                      processed=processed)
```

Comment:
I'm not sure what you'd need this for - and i have quite heavy fears that this will increase the amount of memory needed for hyperopt (hyperopt already has quite heavy memory requirements) - and if the need for memory increases even more it'll become unusable.

Reply:
I don't see way this should create a memory problem, because the data is already there. 
The time it gets free is when the function 'generate_optimizer' is finish. So it is just passed to the loss function before that. So there is no difference in memory usage.
I really need the data in the loss function or is there an other approach?

Second thing what i would need is the config in the loss function. Specially for the fee configuration. Is there a way to access it?

Reply:
Fee is not part of the configuration (unless you use --fee), but is received from the exchange during startup of hyperopt/backtesting (check the `__init__()` method in backtesting).

Reply:
Thanks for your effort :+1: 

I need the fee, because I do not use staking instead I have an all-in/all-out strategy.
So I need to calculate the fees based on the cumulative return.

Reply:
Backtesting (nor hyperopt) does not support compounding at the moment - so your stake-amounts will always be the same, so i'm not quite sure what you'll calculate here...

I'm working on changing this (and it's almost complete - i'm just struggling with a few final calculations i need to adjust to have the result consistent).
so i'm not quite sure why the fee would be relevant (it'll be the same number for each exchange anyway ...)

Reply:
I look up the open/close times of the trades and search these in the btdata. Then I calculate the pct change over all periods and add the fee as penalty to the specific points. After the i calculate the product over all periods and then i get the cumulative return considering all fees.

What do you think about pass the btdata and the hold config object to the loss function?

Reply:
as said, i think you're misjudging what config holds about fees.
fee is NOT in the configuration (at least not the fee that's applied).

It is however available as fee_open and fee_close in the trade results ... so i don't see why this would need config ... ?

Reply:
Sorry for confusion. I don't talk about the config.json. I talk about the self.config object.
Like here: https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/backtesting.py#L105

Reply:
i am aware of this, but the "fee" key is only read - not written to the configuration.
In the default case, where you don't use `--fee 0.0011` to override the exchange default fee - `config['fee']` will be empty, and `self.fee` will have the right value.

https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/backtesting.py#L107

this `self.fee` object is then also assigned to each trade as `open_fee` and `close_fee`, and used/included in profit calculations.

So if you pass in a explicitly high fee, it'll be applied to all profit calculations by freqtrade - so by applying that again in your loss function, you're removing fees twice.

Reply:
Yes, but i don't use the profit calculation at all. Because it does not match to my use case. I set a relative high fee (containing fee + expected slippage) on each run with the cli argument and the i calculate the profit from scratch with the original btdata. So all i need for this is the btdata and the config accessible in the loss function. The fee is not the only config which I want to read in the loss function. I have some loss-function hyperparameters, which i would like to define in the config too. For now that are constants, but that is not ideal for different configurations.